### ISSUE TITLE
new wlanXmon device not found

### ISSUE THREAD

**Issue by ItzYaBoiV** [SilverFoxx/PwnSTAR#8]
Not working now with new airmon-ng update. Says device not found after selecting wireless for ap.

**Comment by THXMASTER** [comment]
yes the new airmon-ng update fcked up many scripts...including this one...please try to fix it.... thanks :)

**Comment by cooljimy84** [comment]
I'm nearly there, just testing it abit and i'll post you the changes to make it work with wlanXmon of the aircrack suite.....

**Comment by ItzYaBoiV** [comment]
Cool man thanks for the update.

**Comment by cooljimy84** [comment]
Ok this might be harder to get the rest going. Change MAC is confusing me, also found it creates a new monX when opening an airodump to scan with.
so far i've got it working on option 2, ish kinda, with a few errors and MAC changing not working. If i can't get any further i'll post up, but don't think it's gonna be a easy one/two line changes

**Comment by ItzYaBoiV** [comment]
ah man take your time no real rush, just a cool script to have. again thanks for the updates.

**Comment by cooljimy84** [comment]
Right so i've spotted a few bits that need correcting to get this working, but i've only tested Grab WPA handshake
It's not pretty and i'm no coder, so i'm sure there are better ways to do what i have done

Open pwnstar in you fav text editor, scroll to line 362
It should start (the BLAH BLAH BLAH is me being lazy and not typeing the whole line)
monap=$ BLAH BLAH BLAH |awk '{print $5"" }'|cut -c -4)
change to
monap=$ BLAH BLAH BLAH |awk '{print $9"" }'|cut -c7-14)
so change the $5 to $9 and change -c -4 to -c7-14

Then
Line 376 through to 389
Should start
y|Y ifconfig $API down && macchanger -A $API && ifconfig $API up

last two lines are (i've done two as the last last line reads the same as the top line)
if [[ "$ap_mac" =~ ^ BLAH BLAH BLAH # checks a sane value has been entered for mac
ifconfig $API down && macchanger -A $API && ifconfig $API up

change all $API entrys to $monap on those 13 lines.

save it and run.
Scanning for APs won't work as you can't create another monitor interface, and ignore errors when changing MAC address, as it does work, even if it says no.

Hope this helps, as i use this script quite a bit aswell

**Comment by expphoto** [comment]
@cooljimy84 . I added your recommendations as a fork to the branch. Let's see what they say!

**Comment by mhtk1122** [comment]
Hi,
Work for me,  4) Simple web server with dnsspoof: redirect the victim to your webpage TESTED
don't work with https (no sslstrip)
Thx

**Comment by eagle1maledetto** [comment]
Sorry to say that, but even after the modify suggested by @cooljimy84 doesn't work for me.

The check in the option 4 goes well, but afterward doesn't work.

Option 3 o 4, same error:

> Starting monitor mode...
> 
> Best to macchange wlan1 and ...
> down: error fetching interface information: Device not found
> down: error fetching interface information: Device not found
> 
> Note: best to start the AP on the same channel as the target
> 
> Do you want to scan eg to discover target channel, ESSID etc? (y/n)
> y
> 
> Starting new monitor interface for scanning...
> 
> Macchanging ...
> 
> down: error fetching interface information: Device not found
> 
> Press enter to close the scan and continue

And doesn't work. It leaves me with an wlan1mon interface.

**Comment by wifiuk** [comment]
does this fix above still work on the latest kali ?

**Comment by CashMoney6980** [comment]
No, it does not work with the most recent version of Kali....unfortunately
